src/Platforms/AbstractPlatform.php

```diff
@@ -3078,10 +3092,25 @@ public function usesSequenceEmulatedIdentityColumns()
         return false;
     }
 
+    /**
+     * Gets the sequence name prefix based on table information.
+     */
+    public function getSequencePrefix(string $tableName, ?string $schemaName = null) : string
```

Comment:
This looks like a leaky abstraction to me. Especially because it's not used by DBAL itself. Given that all the internally used APIs are public, should ORM implement this logic instead?

Reply:
ORM is the one that would be leaking abstraction if calling this method.
Conceptually, ORM is telling DBAL "build me a primary key sequence for this table under this schema". The fact that ORM needs to do more than that is a bad abstraction on its own.
DBAL should defer to platforms that once a PK sequence is requested, a sequence named "bar__foo_seq" should be created if schemas are not supported or "bar.foo_seq" if they are.

I migrated the method to the project that owns it. A later stage would be encapsulate it and handle it accordingly internally without exposing this API.

Reply:
> ORM is the one that would be leaking abstraction if calling this method.

As far as I understand, what you're saying is ORM already relies on implementation details of DBAL. This is a given fact and it cannot be used to make DBAL _leak_ even more of its implementation details.

The very fact that there's a string in ORM that "bar__foo_seq" is built with "__" defined DBAL and "_" defined in ORM looks horrible.

I would agree that the DBAL abstraction used by ORM needs to be _improved_ but the fact that it's already bad cannot be used as a reason to make it even worse.